command.go

```diff
@@ -326,7 +327,7 @@ func (c *Command) RunParallel(ctx context.Context) error {
 				}
 			}()
 			errs <- innerCmd.Run(ctx)
-			c.procs = append(c.procs, innerCmd.procs...)
+			procs <- innerCmd.procs
```

Comment:
the interactions with channels should probably all have contexts to avoid deadlocks

Reply:
I added a select statement to check if the context is done.